### DISCUSSION TITLE
and/or expression line wrapping

### DISCUSSION THREAD

**Discussion by tonycuadra** [JohnnyMorganz/StyLua#571] [General]
I've noticed that long and/or expressions are formatted as:
```lua
(my_very_long_condition)
        and expression_if_true
    or expression_if_false
```

Is the extra indentation of the `and` line intentional (due to other cases involving long conditionals)?

**Comment by JohnnyMorganz** [comment]
Yeah, this is intentional.
We indent based on precedence, and `and` has a higher precedence than `or`. It is explained further in https://github.com/JohnnyMorganz/StyLua/issues/174#issuecomment-854533254


It also highlights the key point that this pseudo-ternary is not an actual ternary. If `expression_if_true` evaluates to false, the whole expression will evaluate to `expression_if_false` regardless of what the condition is!

For Luau, I would recommend switching to if expressions

**Reply by tonycuadra** [reply]
Makes sense! I already did switch these to the safer if/else after the formatting nudge
